src/random_ray/flat_source_domain.cpp

```diff
@@ -861,14 +912,10 @@ void FlatSourceDomain::apply_external_source_to_cell_and_children(
 
 void FlatSourceDomain::count_external_source_regions()
 {
+  n_external_source_regions_ = 0;
 #pragma omp parallel for reduction(+ : n_external_source_regions_)
   for (int sr = 0; sr < n_source_regions_; sr++) {
-    float total = 0.f;
-    for (int e = 0; e < negroups_; e++) {
-      int64_t se = sr * negroups_ + e;
-      total += external_source_[se];
-    }
-    if (total != 0.f) {
```

Comment:
In the new hybrid policy, we make a choice of which estimator to apply based on whether or not there is an external source present in that source region. We want to apply this policy uniformly for all energy groups, even if the source is zero in some groups and positive in others, so we now set a new boolean for the source region in the `FlatSourceDomain::apply_external_source_to_source_region` when a fixed source is being applied. So, we can now just count over this boolean rather than having to sum the source across all groups to see if a source is present or not.